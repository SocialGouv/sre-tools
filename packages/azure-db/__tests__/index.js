const { createDb } = require("../src/createDb");
const { dropDb } = require("../src/dropDb");

jest.mock("execa", (args) =>
  jest.fn((command, args, options) =>
    Promise.resolve({
      stdout: null,
    })
  )
);

describe("Test azure-db commands", () => {
  test("createDb snapshot", async () => {
    const job = await createDb({
      cluster: "cluster-1",
      namespace: "namespace-1",
      database: "database-1",
      user: "user-1",
      password: "password-1",
    });
    expect(job).toMatchSnapshot();
  });

  test("dropDb snapshot", async () => {
    const job = await createDb({
      cluster: "cluster-2",
      namespace: "namespace-2",
      database: "database-2",
      user: "user-2",
    });
    expect(job).toMatchSnapshot();
  });
});
