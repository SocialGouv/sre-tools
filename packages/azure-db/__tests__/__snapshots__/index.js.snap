// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test azure-db commands createDb snapshot 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "name": "sre-tools-create-db-job-50000",
    "namespace": "namespace-1",
  },
  "spec": Object {
    "backoffLimit": 5,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "for-new",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "create-db-user",
            ],
            "env": Array [
              Object {
                "name": "NEW_DB_NAME",
                "value": "database-1",
              },
              Object {
                "name": "NEW_USER",
                "value": "user-1",
              },
              Object {
                "name": "NEW_PASSWORD",
                "value": "password-1",
              },
              Object {
                "name": "NEW_DB_EXTENSIONS",
                "value": "hstore pgcrypto citext uuid-ossp postgis",
              },
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-admin-user",
                },
              },
            ],
            "image": "ghcr.io/socialgouv/docker/azure-db:6.56.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "create-db-user",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`Test azure-db commands dropDb snapshot 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "annotations": Object {
      "kapp.k14s.io/disable-default-label-scoping-rules": "",
      "kapp.k14s.io/disable-default-ownership-label-rules": "",
      "kapp.k14s.io/nonce": "",
      "kapp.k14s.io/update-strategy": "fallback-on-replace",
    },
    "labels": Object {
      "application": "sre-tools",
      "cert": "wildcard",
      "component": "sre-tools",
      "owner": "sre-tools",
      "team": "sre-tools",
    },
    "name": "sre-tools-drop-db-job-50000",
    "namespace": "namespace-2",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "drop-db-user",
            ],
            "env": Array [
              Object {
                "name": "DROP_DATABASE",
                "value": "database-2",
              },
              Object {
                "name": "DROP_USER",
                "value": "user-2",
              },
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-admin-user",
                },
              },
            ],
            "image": "ghcr.io/socialgouv/docker/azure-db:6.56.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "drop-db-user",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;
